docs/source/interpolation.rst

```diff
@@ -131,7 +131,7 @@ proceeds as follows:
 .. code-block:: python3
 
    # First, grab the mesh.
-   m = V.ufl_domain()
+   m = V.extract_unique_domain()
```

Comment:
This is not right, throughout. `extract_unique_domain` is a free function, so you need to:

```
from ufl.domain import extract_unique_domain

...
m = extract_unique_domain(V)
```

To squash further warnings, any use of `expr.ufl_domains()` needs to be replaced with:

```
from ufl.domain import extract_domains

...
domains = extract_domains(expr)
```

Reply:
Thanks, will do.

Reply:
This was useful:

```
function myupdateextractuniquedomain {
  gsed -i 's/base_coords\.extract_unique_domain(/extract_unique_domain(base_coords/g' *.py
  gsed -i 's/assignee\.extract_unique_domain(/extract_unique_domain(assignee/g' *.py
  gsed -i 's/geometric_expr\.extract_unique_domain(/extract_unique_domain(geometric_expr/g' *.py
  gsed -i 's/expr\.extract_unique_domain(/extract_unique_domain(expr/g' *.py
  gsed -i 's/args\[0\]\.extract_unique_domain(/extract_unique_domain(args\[0\]/g' *.py
  gsed -i 's/cxt_kernel\.tensor\.extract_unique_domain(/extract_unique_domain(cxt_kernel\.tensor/g' *.py
  gsed -i 's/builder\.expression\.extract_unique_domain(/extract_unique_domain(builder\.expression/g' *.py
  gsed -i 's/self\.extract_unique_domain(/extract_unique_domain(self/g' *.py
  gsed -i 's/self\.function\.extract_unique_domain(/extract_unique_domain(self\.function/g' *.py
  gsed -i 's/self\.expression\.extract_unique_domain(/extract_unique_domain(self\.expression/g' *.py
  gsed -i 's/slate_tensor\.extract_unique_domain(/extract_unique_domain(slate_tensor/g' *.py
  gsed -i 's/expression\.extract_unique_domain(/extract_unique_domain(expression/g' *.py
  gsed -i 's/_form\.extract_unique_domain(/extract_unique_domain(_form/g' *.py
  gsed -i 's/form\.extract_unique_domain(/extract_unique_domain(form/g' *.py
  gsed -i 's/coeff\.extract_unique_domain(/extract_unique_domain(coeff/g' *.py
  gsed -i 's/coefficient\.extract_unique_domain(/extract_unique_domain(coefficient/g' *.py
  gsed -i 's/coordinates\.extract_unique_domain(/extract_unique_domain(coordinates/g' *.py
  gsed -i 's/coarse\.extract_unique_domain(/extract_unique_domain(coarse/g' *.py
  gsed -i 's/coarse_dual\.extract_unique_domain(/extract_unique_domain(coarse_dual/g' *.py
  gsed -i 's/fine\.extract_unique_domain(/extract_unique_domain(fine/g' *.py
  gsed -i 's/fine_dual\.extract_unique_domain(/extract_unique_domain(fine_dual/g' *.py
  gsed -i 's/test\.extract_unique_domain(/extract_unique_domain(test/g' *.py
  gsed -i 's/Vf\.extract_unique_domain(/extract_unique_domain(Vf/g' *.py
  gsed -i 's/Vc\.extract_unique_domain(/extract_unique_domain(Vc/g' *.py
  gsed -i 's/Pk\.extract_unique_domain(/extract_unique_domain(Pk/g' *.py
  gsed -i 's/it\.extract_unique_domain(/extract_unique_domain(it/g' *.py
  gsed -i 's/op\.extract_unique_domain(/extract_unique_domain(op/g' *.py
  gsed -i 's/F\.extract_unique_domain(/extract_unique_domain(F/g' *.py
  gsed -i 's/V\.extract_unique_domain(/extract_unique_domain(V/g' *.py
  gsed -i 's/J\.extract_unique_domain(/extract_unique_domain(J/g' *.py
  gsed -i 's/Q\.extract_unique_domain(/extract_unique_domain(Q/g' *.py
  gsed -i 's/f\.extract_unique_domain(/extract_unique_domain(f/g' *.py
  gsed -i 's/b\.extract_unique_domain(/extract_unique_domain(b/g' *.py
  gsed -i 's/o\.extract_unique_domain(/extract_unique_domain(o/g' *.py
}
```
If you know how to do the above better, please let me know. I'm eager to learn :)

Reply:
If you go back to the code with ufl domain, then I think the substitution you want is:
```
gsed -i 's/\b(\w+?)\.ufl_domain\(\)/extract_unique_domain(\1)/'
```I think gsed supports non-greedy matching and character classes (otherwise perl definitely does).